rmgpy/rmgobject.pyx

```diff
@@ -82,26 +85,30 @@ cdef class RMGObject(object):
                 obj.make_object(val, class_dict)
                 logging.debug("made object {0}".format(class_name))
                 data[key] = obj
-            elif isinstance(val, list) and val and isinstance(val[0], dict) and 'class' in val[0]:
-                # Call make_object to make a list of objects within the parent object (as in Conformer.Modes)
-                data[key] = list()
-                for entry in val:
-                    class_name = entry['class']
-                    del entry['class']
-                    try:
-                        class_to_make = class_dict[class_name]
-                    except KeyError:
-                        raise KeyError("Class {0} must be provided in the 'class_dict' parameter "
-                                       "to make the object.".format(class_name))
-                    obj = class_to_make()
-                    if class_name in ['LinearRotor', 'NonlinearRotor', 'KRotor', 'SphericalTopRotor', 'HinderedRotor',
-                                      'FreeRotor'] and 'rotationalConstant' in entry and 'inertia' in entry:
-                            # Either `rotationalConstant` or `inertia` should be specified for a rotor.
-                            # Here both are specified, so we delete `inertia`.
-                            del entry['inertia']
-                    obj.make_object(entry, class_dict)
-                    logging.debug("made object {0}".format(class_name))
-                    data[key].append(obj)
+            elif isinstance(val, list) and val:
+                if isinstance(val[0], dict) and 'class' in val[0]:
```

Comment:
I think this section could be combined with the first if clause by adding an elif at the second level. Let me know if that made sense to you.

Reply:
Not so clear. Combine `if isinstance(val[0], dict) and 'class' in val[0]` with `if isinstance(val, dict) and 'class' in val`?

Reply:
Sorry, you can ignore this. I missed the lines that were hidden by GitHub and thought that these changes were part of the same function.